### ISSUE TITLE
File upload is slow. How to control upload chunk size ?

### ISSUE THREAD

**Issue by thedaydreamer** [Lepozepo/S3#101]
This library works great however file upload is a bit slow. Is there any way to control the chunk size or any other way to speed up file upload ?

**Comment by Lepozepo** [comment]
Hmm, I don't have anything set up to control chunk size, let me see if there is anything I can do to enable that. I might take a while though so if you figure it out before I do please submit a PR.

**Comment by gsabran** [comment]
Hi, I'm willing to put some time into getting the multipart upload to work.
I've been struggling to get the right request to send to AWS.

How did you come up with https://github.com/Lepozepo/S3/blob/master/server/sign_request.coffee#L37-L46 ? I've been reading the AWS docs for the multipart upload (http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadUploadPart.html) and I'm not sure how I should presign the urls.

The multipart initiation, cancelation and completion can be done easily server side with the `aws-sdk`

Also how do you see that working with the existing design? Is it ok to add a document in the collection for each part? If I fork and branch, would you have time to look into the stuff until it's ready to do a PR?

**Comment by Lepozepo** [comment]
Hey! We already do multipart upload what's missing is something to control the chunk size during the upload. @gsabran that is the policy that amazon checks to make sure the file that is being uploaded is the file that was signed.

**Comment by gsabran** [comment]
ha where is the multipart upload done? I see only one request been made per file https://github.com/Lepozepo/S3/blob/master/client/functions.coffee#L112

**Comment by Lepozepo** [comment]
Yeah, so that inherently works as a multipart upload, if it wasn't multipart there wouldn't be progress. The docs you were looking at are for multipart server side uploads. I had server methods in a very old iteration of this package that took that into consideration. (it sucks to build that!)

**Comment by gsabran** [comment]
Yes I saw the multipart related code before you removed server side uploading. So it magically happens that AWS will do a multipart upload, like initialize it, splice the file in chunks, upload them and complete the upload as described here http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingRESTAPImpUpload.html ? I thought that for AWS chunks had to be at least 5mb, and I'm receiving progress updates for chunks much smaller that that.
